Lib/test/test_pdb.py

```diff
@@ -2731,6 +2731,46 @@ def test_pdb_issue_gh_103225():
     (Pdb) continue
     """
 
+if not SKIP_ASYNCIO_TESTS:
+    def test_pdb_issue_gh_123321():
```

Comment:
I think this test is failing in the free-threaded build, probably because the initialization of the `_PyOS_ReadlineLock` lock isn't thread-safe without the GIL (and a few other related issues).

That shouldn't be too hard to fix, but we should do it after this PR has landed because the bug you are fixing exists in 3.12, and the free-threading fixes should only go back to 3.13.

In the meantime, you can add a `@support.requires_gil_enabled()` decorator to skip this test in the free-threaded build.

Reply:
Oh thanks, I'll do that. 
I'm changing the tests now and moving them to the readline or builtins module.

Reply:
Was fixed when I took the lock inside the GIL.

I was worried about a potential deadlock, but I don't see any other uses for the lock except this function so it's fine.

Reply:
Actually it does deadlock:

First thread:

`GIL.acq() -> lock.acq() -> GIL.rls() -> |GIL.acq()| -> lock.rls()`

Second thread:
`GIL.acq() -> |lock.acq()| -> GIL.rls() -> GIL.acq() -> lock.rls()`